safekeeper/src/safekeeper.rs

```diff
@@ -511,6 +512,10 @@ pub struct SafeKeeper<CTRL: control_file::Storage, WAL: wal_storage::Storage> {
     pub state: CTRL,               // persistent state storage
 
     pub wal_store: WAL,
+
+    lsn_committed: Sender<Lsn>,
+    lsn_backed_up: Receiver<Lsn>,
+    wal_segment_known: Sender<u32>,
```

Comment:
I can suggest an alternative to adding channels inside SafeKeeper struct. These channels are used for communication with walbackup, and I think we can move this communication to `timeline.rs`. This will follow the current code structure, where we place everything that affects consensus inside `safekeeper.rs` and gluing components together inside `timeline.rs`. 

For example, `commit_lsn` updates when `timeline.rs` calls `process_msg` and then it gets `AppendResponse`. In that `AppendResponse` we have latest `commit_lsn`, so `timeline.rs` can notify `send_wal.rs` that `commit_lsn` has just updated (that happens inside `notify_wal_senders`). So we can remove `lsn_committed` channel and just modify existing `notify_wal_senders` to notify walbackup.

`lsn_backed_up` can be replaced with just calling some `SafeKeeper` method, or directly persisting update to `safekeeper.state`.

When we creating/restoring timeline, we usually already know `wal_seg_size`, because it's persisted to control file. If not, we can wait for Greeting message in `process_msg` inside `timeline.rs`. Once we know `wal_seg_size`, we can pass it to walbackup (or initialize walbackup at this moment, see below).

One way to initialize walbackup inside `timeline.rs` is to place `walbackup: Option<WalBackup>` inside `SharedState`, and create it only when we know `wal_seg_size`.

Reply:
I think we should discuss this proposal over a call.
looking at the code structure either I'm confused or the implementation would be more convoluted